redaxo/src/addons/install/lib/install.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+/**
+ * @package redaxo\install
+ */
+class rex_install
+{
+    /**
+     * Download an AddOn from redaxo.org.
+     *
+     * @param string $addonKey e.g. "yform"
+     * @param string $version  e.g. "3.2.1"
+     *
+     * @throws rex_exception
+     */
+    public function downloadAddon(string $addonKey, string $version): void
+    {
+        if (rex_addon::exists($addonKey)) {
+            throw new rex_exception(sprintf('AddOn "%s" already exists!', $addonKey));
+        }
+
+        $packages = rex_install_packages::getAddPackages();
+        if (!isset($packages[$addonKey])) {
+            throw new rex_exception(sprintf('AddOn "%s" does not exist!', $addonKey));
+        }
+        $package = $packages[$addonKey];
+        $files = $package['files'];
+
+        // search fileId by version
+        $fileId = null;
+        foreach ($files as $fId => $fileMeta) {
+            if ($fileMeta['version'] !== $version) {
+                continue;
+            }
+            $fileId = $fId;
+            break;
+        }
+
+        if (!$fileId || !isset($files[$fileId])) {
+            throw new rex_exception(sprintf('Version "%s" not found!', $version));
+        }
+
+        $install = new rex_install_package_add();
+        $message = $install->run($addonKey, $fileId);
+
+        if ('' !== $message) {
+            throw new rex_exception($message);
+        }
+    }
+
+    /**
+     * Updates an AddOn from redaxo.org.
```

Comment:
Update heißt es muss bereits installiert sein?

Ist update immer auf eine neuere version oder geht das hier auch zurück auf ältere?

Reply:
es muss nicht installiert sein.
Wie auch beim normalen Installer im backend, kann immer nur höher gegangen werden. 

siehe
https://github.com/redaxo/redaxo/blob/e8e124892e82bd7792e44a592dab85c5c5020588/redaxo/src/addons/install/lib/package/package_update.php#L31-L33

Reply:
diese vorbedingungen/rahmenbedingungen bitte im phpdoc beschreiben

Reply:
```suggestion
     * Updates an AddOn from redaxo.org.
     * You can only update downloaded addons. It does not matter whether they are installed or not. You can only upgrade to newer versions.
```

gerne verbessern / korrigieren